app/src/main/java/org/mozilla/fenix/perf/HotStartPerformanceMonitor.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.os.SystemClock
+import android.util.Log
+
+/**
+ * Monitors and reports elapsed time to complete hot startup of an Activity. Callers are expected
+ * to call the appropriate lifecycle methods at the appropriate times.
+ *
+ * A "hot startup" is when the application moves from the background to the foreground and both the
+ * Application/process and the Activity under measurement are already created.
+ *
+ * Unfortunately, this monitor does not capture the entire duration of hot start because the
+ * framework calls several Android framework methods in the application's process before we are able
+ * to add monitoring code (i.e. the first time our application code is called in onRestart). An
+ * alternative implementation could measure performance from outside the application.
+ *
+ * To see logs from this class, the user must enable VERBOSE logging for the appropriate tag:
+ *   adb shell setprop log.tag.FenixPerf VERBOSE
+ */
+class HotStartPerformanceMonitor(
+    private val log: (String) -> Unit = { Log.v(Performance.TAG, it) }, // android log to minimize overhead.
```

Comment:
You do *not* have to do this, but I would have called this something like "report" because it indicates that we are reporting the hot start time.

Reply:
Despite the way we're using it (to report), this method still functionally acts as a generic logging method so I'm inclined to leave as it is.

fwiw, if I did, `{ Log.v(Performance.TAG, "hot start: $it") }`, then a name like `reportHotStart` would make sense to me. :)